### PR TITLE
fix(elasticsearch): Domain.fromDomainAttributes gives "Invalid URL" when endpoint is a token

### LINKED ISSUE #15188: (elasticsearch): Domain.fromDomainAttributes gives "Invalid URL" when endpoint is a token
I'm getting a `Invalid URL` when trying to use `Domain.fromDomainAttributes` with a CfnDomain object. The CfnDomain object was retrieved from a CfnInclude object. Here's the snippet of code that's failing for me.

```
private getDomainFromTemplate(name: string, cfnTemplate: CfnInclude): IDomain {
    const cfnDomain = cfnTemplate.getResource(name) as CfnDomain;
    return Domain.fromDomainAttributes(this, name, {domainArn: cfnDomain.attrArn, domainEndpoint: cfnDomain.attrDomainEndpoint});
}
```

### Reproduction Steps

Here's some minimal code that causes the bug.
```
import * as elasticsearch from '@aws-cdk/aws-elasticsearch';

const domain = new elasticsearch.CfnDomain(this, "domain", {
    domainName: "domainName"
});
elasticsearch.Domain.fromDomainAttributes(this, name, {domainArn: domain.attrArn, domainEndpoint: domain.attrDomainEndpoint});
```

### What did you expect to happen?

The error shouldn't happen and the Domain object should be successfully created.

### What actually happened?

```
internal/url.js:257
  throw new ERR_INVALID_URL(input);
  ^

TypeError [ERR_INVALID_URL]: Invalid URL: ${Token[TOKEN.1766]}
    at onParseError (internal/url.js:257:9)
    at new URL (internal/url.js:333:5)
    at extractNameFromEndpoint (/Users/saimuty/workplace/dr_infra/src/JuryDisputeResolutionNAWSCDK/node_modules/monocdk/lib/aws-elasticsearch/lib/domain.js:946:26)
    at Function.fromDomainAttributes (/Users/saimuty/workplace/dr_infra/src/JuryDisputeResolutionNAWSCDK/node_modules/monocdk/lib/aws-elasticsearch/lib/domain.js:903:28)
```

### Environment

  - **CDK CLI Version  :** 1.107.0 (build 52c4434)
  - **Framework Version:** 1.107.0 (build 52c4434)
  - **Node.js Version:** v12.21.0
  - **OS               :** MacOS Big Sur 11.4
  - **Language (Version):** Typescript (4.0.5)

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by HelioStrike**
`Domain.fromDomainAttributes` throws the error "Invalid URL" when a token endpoint is provided. In this PR, the domain name is retrieved from the scope using the arn present in `DomainAttributes`, and if no name is found, we try to get it from the endpoint (as it has been done so far). Also added a test to verify that the error disappears with this change, even when it is given token attributes.

closes #15188

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by BenChaimberg** [comment]
Please add a summary description of the changes contained in this PR

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts [hunk 1]
```diff
@@ -913,6 +913,15 @@ describe('import', () => {
     expect(stack).not.toHaveResource('AWS::Elasticsearch::Domain');
   });
 
+  test('static fromDomainAttributes(attributes) with token arn and endpoint does not throw validation error', () => {
+    const domainArn = new CfnParameter(stack, 'domainArn', { type: 'String' }).valueAsString;
+    const domainEndpoint = new CfnParameter(stack, 'domainEndpoint', { type: 'String' }).valueAsString;
+
+    expect(() => {
+      Domain.fromDomainAttributes(stack, 'Domain', { domainArn, domainEndpoint });
+    }).not.toThrowError(/Invalid URL/);
```
Comment (BenChaimberg) [inline packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts:922]: I would prefer if the verification in this test was similar to the previous tests (confirming name, ARN, and endpoint are concrete values)

**Review by BenChaimberg — CHANGES_REQUESTED** [review]
the modes of these files were changed to allow execution; please revert

**Comment by HelioStrike** [comment]
@BenChaimberg Updated the PR. Had to change modes due to some permission issue with yarn. Reverted it now, and also modified tests to check equality.
Also, when modifying a PR, should I force push modified commits, or add a commit on top?

**Comment by HelioStrike** [comment]
@BenChaimberg Updated the PR. Had to change modes due to some permission issue with yarn. Reverted it now, and also modified tests to check equality.
Also, when modifying a PR, should I force push modified commits, or add a commit on top?

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts (hunk 1 above)
Comment (HelioStrike) [reply packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts]: Checked equality with the CfnParameters. I'm new to this repo, so not sure how to turn concrete values into Tokens and then use them.

**Comment by BenChaimberg** [comment]
Please commit on top, as Github doesn't have great support for diffing between force-pushed commits

File: packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts (hunk 1 above)
Comment (BenChaimberg) [reply packages/@aws-cdk/aws-elasticsearch/test/domain.test.ts]: yep this is what I wanted!

**Review by BenChaimberg — APPROVED** [review]

**Pull request merged**